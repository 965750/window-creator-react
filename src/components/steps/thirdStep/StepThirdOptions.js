import React, { Component } from 'react'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import ColorBox from './ColorBox'

class StepThirdOptions extends Component {
    render() {
        const colorBoxes = this.props.colorBoxes.map(box => (
            <ColorBox classes="w-1/4" key={box.id} data={box} />
        ))

        return (
            <div className="pt-2">
                <div className="mb-10">
                    <p className="mb-4 text-darkText border-b border-mercury">
                        <FormattedMessage
                            id="Choose color"
                            defaultMessage="Choose color"
                        />
                    </p>
                    <div className="flex">{colorBoxes}</div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    colorBoxes: state.creator.colorBoxes,
})

export default connect(mapStateToProps)(StepThirdOptions)
