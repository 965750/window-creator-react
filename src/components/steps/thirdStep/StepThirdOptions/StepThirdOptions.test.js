import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import StepThirdOptions from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
  creator: {
    colorBoxes: [{
      id: 'mockId',
      color: 'red',
      active: false,
      label: 'Black'
    }],
  }
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('StepThirdOptions', () => {
  let wrapper
  const initialProps = {
    data: {
      id: 'mockId',
      color: 'red',
      active: false,
      label: 'Black'
    }
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <StepThirdOptions {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
