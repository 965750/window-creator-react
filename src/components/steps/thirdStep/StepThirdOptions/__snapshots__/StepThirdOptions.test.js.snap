// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepThirdOptions should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-2"
  >
    <div
      class="mb-30"
    >
      <p
        class="mb-4 text-osloGray border-b border-mercury"
      >
        Wybierz kolor
      </p>
      <div
        class="flex"
      >
        <button
          class="w-1/3 cursor-pointer"
          data-testid="colorBoxWrapper"
        >
          <div
            class="colorBox mx-auto rounded-full mb-1"
            style="background: red;"
          />
          <div
            class="flex justify-center"
          >
            <div
              class="radio__wrapper border-botticelli border-4 rounded-full relative mt-sm"
              data-testid="radioWrapper"
            />
            <span
              class="ml-1"
            >
              Czarny
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
